features/extensions/developer_uses_extension.feature

```diff
@@ -24,9 +24,16 @@ Feature: Developer uses extension
          */
         public function load(ServiceContainer $container)
         {
+            // Add new matcher to container
             $container->set('matchers.seven', function (ServiceContainer $c) {
                 return new BeSevenMatcher($c->get('formatter.presenter'));
             });
+
+            // Add new matcher to list of matchers in container
+            $matchers = $container->get('phpspec.matchers');
```

Comment:
this is not a suitable way to extend an array of matchers. It breaks lazyness of definitions, which will then forbid overriding any dependency of these matchers after this particular extension was registered.

You need to keep a clear distinction between the configuration phase of the container (where you define service definitions) and the runtime phase (where you can retrieve services from the container).
Mixing them together leads to broken behaviors and very hard to debug issues (and the container defined in PhpSpec does not implement any safeguard to warn you, as it was inspired by Pimple 1 which suffered from the same limitations).